src/Interpreters/Session.cpp

```diff
@@ -252,6 +251,28 @@ class NamedSessionsStorage
 
                 const auto & session = session_it->second;
 
+                /// We can get here only if the session is still in use somehow. But since we don't allow concurrent usage
+                /// of the same session, that means that the session wasn't yet released or closed.
+                //
+                /// There are 3 ways that could lead to such a situation in theory:
+                ///   1. The session is released, but being reused before the timeout expires
+                ///   2. The session was closed, but a new one with the same name was created before the timeout for a
+                ///      previous one expires.
+                ///   3. The session was released, but the pointer wasn't reset yet.
```

Comment:
> The session was released, but the pointer wasn't reset yet

It looks like only this line from this big comment is still relevant. Let's keep only this line and move it inside the block `if (session.use_count() != 1)` to clarify that this comment is related to that condition.